src/combinatorics/simple_graph/connectivity/subgraph.lean

```diff
@@ -0,0 +1,201 @@
+/-
+Copyright (c) 2023 Kyle Miller, Rémi Bottinelli. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kyle Miller, Rémi Bottinelli
+-/
+
+import combinatorics.simple_graph.basic
+import combinatorics.simple_graph.connectivity
+import combinatorics.simple_graph.subgraph
+/-!
+# Connectivity of subgraphs
+-/
+
+universes u v
+
+namespace simple_graph
+
+variables {V : Type u} {V' : Type v} {G : simple_graph V} {G' : simple_graph V'}
+
+/-- A subgraph is connected if it is connected as a simple graph. -/
+@[reducible] def subgraph.connected (H : G.subgraph) : Prop := H.coe.connected
+
+lemma subgraph.connected_iff (H : G.subgraph) :
+  H.connected ↔ H.coe.preconnected ∧ H.verts.nonempty :=
+begin
+  change H.coe.connected ↔ _,
+  rw [connected_iff, set.nonempty_coe_sort],
+end
+
+lemma singleton_subgraph_connected {v : V} : (G.singleton_subgraph v).connected :=
+begin
+  split,
+  rintros ⟨a, ha⟩ ⟨b, hb⟩,
+  simp only [singleton_subgraph_verts, set.mem_singleton_iff] at ha hb,
+  subst_vars
+end
+
+@[simp] lemma subgraph_of_adj_connected {v w : V} (hvw : G.adj v w) :
+  (G.subgraph_of_adj hvw).connected :=
+begin
+  split,
+  rintro ⟨a, ha⟩ ⟨b, hb⟩,
+  simp only [subgraph_of_adj_verts, set.mem_insert_iff, set.mem_singleton_iff] at ha hb,
+  obtain (rfl|rfl) := ha; obtain (rfl|rfl) := hb;
+    refl <|> { apply adj.reachable, simp },
+end
+
+lemma induce_singleton_connected (v : V) :
+  (G.induce {v}).connected :=
```

Comment:
Not sure what's the point of this lemma.

Reply:
I think it was used at some point, and got replaced by `singleton_subgraph_connected`. Is there a point to removing it, though?

Reply:
It's a straightforward combination of two existing lemmas.

Reply:
Alright, I'll drop it then!